pkg/operators/kubeipresolver/kubeipresolver.go

```diff
@@ -134,6 +145,179 @@ func (m *KubeIPResolverInstance) EnrichEvent(ev any) error {
+				}
+				k8sKindAcc, err := k8sSubAcc.AddSubField("kind", api.Kind_String,
+					datasource.WithAnnotations(map[string]string{
+						datasource.ColumnsMaxWidthAnnotation: "12",
+					}),
+					datasource.WithFlags(datasource.FieldFlagHidden),
+				)
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "kind", err)
+				}
+				k8sNameAcc, err := k8sSubAcc.AddSubField("name",
+					api.Kind_String,
+					datasource.WithAnnotations(map[string]string{
+						datasource.TemplateAnnotation: "pod",
+					}),
+					datasource.WithFlags(datasource.FieldFlagHidden),
+				)
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "name", err)
+				}
+				k8sNamespaceAcc, err := k8sSubAcc.AddSubField("namespace",
+					api.Kind_String,
+					datasource.WithAnnotations(map[string]string{
+						datasource.TemplateAnnotation: "namespace",
+					}),
+					datasource.WithFlags(datasource.FieldFlagHidden),
+				)
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "namespace", err)
+				}
+				k8sLabelsAcc, err := k8sSubAcc.AddSubField("labels", api.Kind_String, datasource.WithFlags(datasource.FieldFlagHidden))
+				if err != nil {
+					return nil, fmt.Errorf("adding field %q: %w", "labels", err)
+				}
+				ea := endpointAccessors{
+					root:            ep,
+					subK8sKind:      k8sKindAcc,
+					subK8sName:      k8sNameAcc,
+					subK8sNamespace: k8sNamespaceAcc,
+					subK8sLabels:    k8sLabelsAcc,
+				}
+				epAccessors[ds] = append(epAccessors[ds], ea)
+			}
+		}
+	}
+
+	// No endpoints found, nothing to do
+	if len(epAccessors) == 0 {
+		return nil, nil
+	}
+
+	k8sInventory, err := common.GetK8sInventoryCache()
+	if err != nil {
+		return nil, fmt.Errorf("creating k8s inventory cache: %w", err)
+	}
+
+	return &KubeIPResolverInstance{
+		k8sInventory:       k8sInventory,
+		endpointsAccessors: epAccessors,
+	}, nil
+}
+
+func (k *KubeIPResolver) Priority() int {
+	return Priority
+}
+
+func (m *KubeIPResolverInstance) PreStart(gadgetCtx operators.GadgetContext) error {
+	m.k8sInventory.Start()
+	return nil
+}
+
+func (m *KubeIPResolverInstance) Start(gadgetCtx operators.GadgetContext) error {
+	for ds, acc := range m.endpointsAccessors {
+		ds.Subscribe(func(source datasource.DataSource, data datasource.Data) error {
+			var errs error
+			for _, a := range acc {
+				ips := a.root.GetSubFieldsWithTag("type:" + ipAddrType)
+				if len(ips) != 1 {
+					errors.Join(errs, fmt.Errorf("%s: expected %d %q field, got %d", a.root.Name(), 1, ipAddrType, len(ips)))
+					continue
+				}
+				if ips[0].Size() != 16 {
+					errors.Join(errs, fmt.Errorf("%s: expected %q field to have size %d, got %d", a.root.Name(), ipAddrType, 16, ips[0].Size()))
+					continue
+				}
+
+				version := a.root.GetSubFieldsWithTag("name:version")
+				if len(version) != 1 {
+					errors.Join(errs, fmt.Errorf("%s: expected %d %q field, got %d", a.root.Name(), 1, "version", len(version)))
+					continue
+				}
```

Comment:
These checks can be already done on `InstantiateDataOperator` and return the error there, not every time the ds emits data.

Here, you can trust that:

```suggestion
				ip := a.root.GetSubFieldsWithTag("type:" + ipAddrType)[0]
				version := a.root.GetSubFieldsWithTag("name:version")[0]
```